cmd/lakefs/cmd/run.go

```diff
@@ -181,6 +184,48 @@ var runCmd = &cobra.Command{
 	},
 }
 
+// A foreign repo is a repository which namespace doesn't match the current block adapter.
+// A foreign repo might exists if the lakeFS instance configuration changed after a repository was
+// already created. The behaviour of lakeFS for foreign repos is undefined and should be blocked.
+func checkForeignRepos(ctx context.Context, logger logging.Logger, authMetadataManager *auth.DBMetadataManager, blockStore block.Adapter, c *catalog.Catalog) {
+	initialized, err := authMetadataManager.IsInitialized(ctx)
+	if err != nil {
+		logger.WithError(err).Fatal("Failed to read setup time")
+	}
+	if initialized {
```

Comment:
Can you add some logs?  E.g. emit one of these:
* `failed to check if initialized`
* `not initialized, not checking repos`
* `initialized, checking to see repos match block adapter xyzzy`

Also, why won't it work before setup?  I'd expect `listRepositories` to exit with 0 read, meaning the loop ends.  No?

Reply:
Added the logs.
It wouldn't work because the tables don't exist just yet, so it's a different type of error than ErrNoRows which returns on 0 results.